File: shared/enums.ts
// Shared enum constants — single source of truth for frontend and backend
// Backend (C#): OscApi.Models.Enums mirrors these values exactly

export const TICKET_CATEGORIES = [
  'general_inquiry',
  'procedure_query',
  'application_support',
  'license_delay',
  'complaint',
  'vip',
] as const;
export type TicketCategory = (typeof TICKET_CATEGORIES)[number];

export const TICKET_STATUSES = [
  'New',
  'Assigned',
  'InProgress',
  'PendingExternal',
  'Resolved',
  'Closed',
] as const;
export type TicketStatus = (typeof TICKET_STATUSES)[number];

export const TICKET_PRIORITIES = ['Low', 'Medium', 'High', 'Critical'] as const;
export type TicketPriority = (typeof TICKET_PRIORITIES)[number];

export const AUTHOR_ROLES = ['Investor', 'Officer', 'System'] as const;
export type AuthorRole = (typeof AUTHOR_ROLES)[number];

export const CHAT_LANGUAGES = ['En', 'Fr', 'Ar', 'Zh', 'Sw'] as const;
export type ChatLanguage = (typeof CHAT_LANGUAGES)[number];

export const CHAT_SENTIMENTS = ['Positive', 'Neutral', 'Negative'] as const;
export type ChatSentiment = (typeof CHAT_SENTIMENTS)[number];

export const CHAT_TIERS = ['Ai', 'Kb', 'Suggestions'] as const;
export type ChatTier = (typeof CHAT_TIERS)[number];

export const INVESTOR_TYPES = ['Individual', 'Institutional', 'Foreign'] as const;
export type InvestorType = (typeof INVESTOR_TYPES)[number];

export const SLA_HOURS: Record<TicketCategory, number> = {
  general_inquiry: 24,
  procedure_query: 8,
  application_support: 4,
  license_delay: 2,
  complaint: 2,
  vip: 1,
};

File: backend/src/OscApi/Models/Enums.cs
namespace OscApi.Models;

public enum TicketCategory
{
    GeneralInquiry,
    ProcedureQuery,
    ApplicationSupport,
    LicenseDelay,
    Complaint,
    Vip
}

public enum TicketPriority
{
    Low,
    Medium,
    High,
    Critical
}

public enum TicketStatus
{
    New,
    Assigned,
    InProgress,
    PendingExternal,
    Resolved,
    Closed
}

public enum AuthorRole
{
    Investor,
    Officer,
    System
}

public enum ChatLanguage
{
    En,
    Fr,
    Ar,
    Zh,
    Sw
}

public enum ChatSentiment
{
    Positive,
    Neutral,
    Negative
}

public enum ChatTier
{
    Ai,
    Kb,
    Suggestions,
    Escalation
}

public enum InvestorType
{
    Individual,
    Institutional,
    Foreign
}

public enum InvestorExperience
{
    Beginner,
    Intermediate,
    Advanced
}

public enum InvestmentGoal
{
    Growth,
    Income,
    Diversification,
    Strategic
}

public enum TimeHorizon
{
    ShortTerm,
    MediumTerm,
    LongTerm
}

public enum RiskTolerance
{
    Conservative,
    Moderate,
    Aggressive
}

public enum CapitalSource
{
    Savings,
    Loan,
    Partnership,
    Grant
}

public enum InvestorTimeframe
{
    Immediate,
    ThreeMonths,
    SixMonths,
    OneYear
}

public enum InvestorStatus
{
    New,
    Contacted,
    Active,
    Inactive
}

/// <summary>
/// A business registration's own lifecycle — distinct from <see cref="TicketStatus"/>
/// so a registration is a tracked transaction with its own state, not a support
/// ticket wearing a registration form's clothing.
/// </summary>
public enum BusinessRegistrationStatus
{
    Received,
    UnderReview,
    NameApproved,
    NameRejected,
    CertificateIssued,
    Rejected
}
